iroh-gossip/src/proto/util.rs

```diff
@@ -0,0 +1,182 @@
+use rand::{
```

Comment:
missing unit tests, ideally some proptests for the pieces in this file

Reply:
I added basic unit tests for `IndexSet`, `TimerMap` and the new `base32` utils.
What parts would you want to proptest? Can do, just not super sure if/where it would make sense here.